include/cppkafka/message.h

```diff
@@ -127,6 +127,27 @@ class CPPKAFKA_API Message {
     }
     
 #if (RD_KAFKA_VERSION >= RD_KAFKA_HEADERS_SUPPORT_VERSION)
+    /**
+     * \brief Sets the message's header list.
+     * \note After this call, the Message will take ownership of the header list.
+     */
+    void set_header_list(const HeaderListType& headers) {
```

Comment:
I don't think this overload is a good idea, at least with the current semantics. If a function takes something by const ref, it should by no means be allowed to take ownership of it. The overload below that takes an rvalue-ref makes sense, as in that case someone specifically moved whatever they were passing. 

Can't this overload just copy the headers?

Reply:
Well, if you look at the code, I actually assert that it's non-owning. There are instances of `HeaderListType` which are non-owning by default (like the one inside a `Message` class) so in this case passing it by `const` is right. If you only use the rvalue method that means you are moving the handle out, which is not always the case. I can remove this methond (and the `is_owning`) if you like, but consider the above scenario and lmk.

Reply:
So currently the only place where there's non owning headers is in a `Message`, right? This means that the only way to use this overload is if you grab a produced message, let's call it A, and message B which for example just failed to be produced and you want to re-produce it using the headers from message A. Am I missing something? 

You can only use this overload if you pull non owning headers from somewhere and those currently are only inside a Message. Either that or someone explicitly created a `rd_kafka_headers_t` and built a non owning cppkafka wrapper around it and now want to use it.

Given these use cases (please correct me if I'm missing something) I think:
* The use case of pulling headers from a message and putting them into another one is not common at all so people can just use the other overload.
* We shouldn't be encouraging people to use non owning handles because they make everything more complicated. It's better if they stick to what cppkafka provides or if there's extra functionality missing, the library should provide it.

I'd even go ahead and merge both overloads into one that simply takes a `HeaderListType` by value. If you want to move your own in there, do it, otherwise it gets copied.

Reply:
Ok I changed the overload so that when passing by `const HeaderList&` we actually make a copy via `rd_kafka_header_copy` and don't enfoce ownership. Merging the two and passing it by value does not guarantee that the _copy_ will actually contain a copied list. If the HeaderList is non-owning, the copy is nothing more than copying the handle around, which we then pass ownership to rdkafka, so whoever calls this setter method will get shot in the foot w/o knowing. 
Yes, in reference to your comment, the only non-owning instance is inside Message, however we can also make `MessageBuilder` non-owning as per my comments regarding the excessive copies. Also you never know what the users might do. They might call `get_handle()` on a message builder or message, make a new non-owning HeaderList and then pass that around. So we cannot assume that only a Message contains non-owning headers.